client/src/components/AccountPrompt/index.js

```diff
@@ -0,0 +1,92 @@
+import React, { useState } from 'react'
+import PropTypes from 'prop-types'
+import { useHistory } from 'react-router-dom'
+
+import { useConfiguration } from '../../providers/Configuration'
+import Button from '../Button'
+import {
+  Form,
+  FormContainer,
+  Input,
+  PromptLink,
+  PromptContainer,
+} from './styled'
+import { H2 } from '../../styles/Text'
+import { login, useAuth } from '../../auth'
+
+const AccountPrompt = ({ text, page, pageHeader, pageText, link }) => {
+  const [isLoading, setIsLoading] = useState(false)
+  const [username, setUsername] = useState('')
+  const [password, setPassword] = useState('')
+  const { apiUrl } = useConfiguration()
+  const history = useHistory()
+  const [logged] = useAuth()
+
+  const isFormInvalid = !username || !password
+
+  async function onSubmit(e) {
+    e.preventDefault()
+    setIsLoading(true)
+    const body = {
+      username,
+      password,
+    }
+    try {
+      const response = await fetch(`${apiUrl}/${page}`, {
```

Comment:
this prop should be named in a way that describes what endpoint you're hitting. maybe `apiEndpoint`